workshops/views.py

```diff
@@ -1672,48 +1683,57 @@ def instructors(request):
                 # not any lesson within the list (as it would be with
                 # `.filter(lessons_in=lessons)`)
                 for lesson in lessons:
-                    instructors = instructors.filter(
+                    people = people.filter(
                         qualification__lesson=lesson
                     )
 
             if data['airport']:
                 x = data['airport'].latitude
                 y = data['airport'].longitude
                 # using Euclidean distance just because it's faster and easier
-                complex_F = ((F('airport__latitude') - x) ** 2
-                             + (F('airport__longitude') - y) ** 2)
-                instructors = instructors.annotate(distance=complex_F) \
-                                         .order_by('distance', 'family')
-
-            if data['latitude'] and data['longitude']:
-                x = data['latitude']
-                y = data['longitude']
-                # using Euclidean distance just because it's faster and easier
-                complex_F = ((F('airport__latitude') - x) ** 2
-                             + (F('airport__longitude') - y) ** 2)
-                instructors = instructors.annotate(distance=complex_F) \
-                                         .order_by('distance', 'family')
+                complex_F = ((F('airport__latitude') - x) ** 2 +
+                             (F('airport__longitude') - y) ** 2)
+                people = people.annotate(distance=complex_F) \
+                               .order_by('distance', 'family')
 
             if data['country']:
-                instructors = instructors.filter(
+                people = people.filter(
                     airport__country__in=data['country']
                 ).order_by('family')
 
             if data['gender']:
-                instructors = instructors.filter(gender=data['gender'])
+                people = people.filter(gender=data['gender'])
 
             if data['instructor_badges']:
                 for badge in data['instructor_badges']:
-                    instructors = instructors.filter(badges__name=badge)
+                    people = people.filter(badges__name=badge)
+
+            # it's faster to count role=helper occurences than to check if user
+            # had a role=helper
+            if data['was_helper']:
+                people = people.filter(num_helper__gte=1)
+
+            if data['was_organizer']:
+                people = people.filter(num_organizer__gte=1)
+
+            if data['is_in_progress_trainee']:
+                people = (
+                    people.filter(task__event__tags=TTT,
+                                  task__role__name='learner')
+                    .exclude(task__event__tags=stalled)
+                    .exclude(badges__in=instructor_badges)
+                )
```

Comment:
This excludes a person who is not a certified instructor, attended one stalled TTT event and one non-stalled TTT event.